pkg/util/gziphandler/gzip.go

```diff
@@ -322,21 +324,17 @@ func GzipHandlerWithOpts(opts ...option) (func(http.Handler) http.Handler, error
 
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Add(vary, acceptEncoding)
-			if acceptsGzip(r) {
+			if acceptsGzip, acceptsIdentity := requestAcceptance(r); acceptsGzip {
 				gw := &GzipResponseWriter{
-					ResponseWriter: w,
-					index:          index,
-					minSize:        c.minSize,
-					contentTypes:   c.contentTypes,
+					ResponseWriter:  w,
+					index:           index,
+					minSize:         c.minSize,
+					contentTypes:    c.contentTypes,
+					acceptsIdentity: acceptsIdentity,
 				}
 				defer gw.Close()
 
-				if _, ok := w.(http.CloseNotifier); ok {
```

Comment:
From the `http.CloseNotifier` comment:

```go
// Deprecated: the CloseNotifier interface predates Go's context package.
// New code should use Request.Context instead.
```

This was here just because the source library has been unmaintained for years, but when I brought the code here, linter complained about it.